lib/functions/ProofUtil.ts

```diff
@@ -35,13 +53,23 @@ function partiallyValidateJWS(jws: string): void {
   }
 }
 
-function setJWSDefaults(args: JWTSignerArgs, issuerUrl: string, clientId?: string): JWTSignerArgs {
+function setJWSDefaults(
+  args: ProofOfPossessionCallbackArgs,
+  endpointMetadata: EndpointMetadata,
+  clientId?: string
+): { header: JWTHeader; payload: JWTPayload } {
   const now = +new Date();
-  const aud = args.payload.aud ? args.payload.aud : issuerUrl;
+  if (!endpointMetadata) {
+    throw new Error('No endpointMetadata provided');
+  }
+  const aud = endpointMetadata.issuer ? endpointMetadata.issuer : args.payload.aud;
```

Comment:
I would expect this the other way around. endpointMetadata always contains an issuer value. 
That should be the default. But if you do supply an issuer as argument use that

Reply:
changed the order in ternary